core/src/main/java/org/apache/carbondata/core/metadata/schema/PartitionInfo.java

```diff
@@ -65,6 +65,31 @@ public PartitionInfo(List<ColumnSchema> columnSchemaList, PartitionType partitio
     this.partitionIds = new ArrayList<>();
   }
 
+  /**
+   * add partition means split default partition, add in last directly
+   */
+  public void  addPartition(int addPartitionCount) {
+    for (int i = 0; i < addPartitionCount; i++) {
+      partitionIds.add(++MAX_PARTITION);
+      numPartitions++;
+    }
+  }
+
+  /**
+   * e.g. original partition[0,1,2,3,4,5]
+   * split partition 2 to partition 6,7,8 (will not reuse 2)
+   * then sourcePartitionId is 2, newPartitionNumbers is 3
+   * @param sourcePartitionIndex
+   * @param newPartitionNumbers
+   */
+  public void splitPartition(int sourcePartitionIndex, int newPartitionNumbers) {
```

Comment:
When will partition-split action be triggered？

In my memory, there is no “partition-split” action in Hive, etc.